realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -117,6 +119,19 @@ private RealmCache(RealmConfiguration config) {
 
         RefAndCount refAndCount = cache.refAndCountMap.get(RealmCacheType.valueOf(realmClass));
 
+        if (refAndCount.globalCount == 0) {
+            SharedRealm sharedRealm = SharedRealm.getInstance(configuration);
+            if (Table.primaryKeyTableNeedsMigration(sharedRealm)) {
```

Comment:
`An empty transaction` will block/be blocked other thread/process.
eg. there is a long running transaction in a background process, the Realm in this process cannot be opened until that transaction finished. We got some similar complains before.